core/src/main/java/org/apache/iceberg/MergingSnapshotUpdate.java

```diff
@@ -266,6 +235,42 @@ protected void add(DataFile file) {
     }
   }
 
+  private void groupManifestsByPartitionSpec(
+      Snapshot current,
+      Map<Integer, List<ManifestFile>> groups,
+      StrictMetricsEvaluator metricsEvaluator, Set<CharSequenceWrapper> deletedFiles) throws IOException {
```

Comment:
I think this should be divided into 2 methods:
* `filterManifests` that runs all the filters, and
* `groupManifestsByPartitionSpec` that does the grouping

The current name doesn't indicate that filtering is also done and mixes the two concerns.